### PR TITLE
Notes lecture 6

### PR DESCRIPTION
**Pull request by mathildebadoual**
I changed the size of the the figures and added a python file in /notes which creates the figures.

### CODE REVIEW

**Comment by mrtzh** [comment]
Thanks. The font now looks even smaller to me relative to the size of the figure. Let's wait to see how it integrates into the pdf. It's probably best to adjust it once you know where it's supposed to fit in.

**Comment by mathildebadoual** [comment]
Here is the final lecture 6.

**Comment by mrtzh** [comment]
Sorry, but I'm not seeing the tex file. Also, again, the font size on the figures looks tiny to me.

**Comment by mathildebadoual** [comment]
Sorry, I think now everything works.

File: notes/lecture6.tex [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+In this lecture, we seek to find methods that converge faster than those discussed in previous lectures. To derive this accelerated method, we start by considering the special case of optimizing quadratic functions.
+
+\subsection{Quadratics}
+
+\begin{definition}[Quadratic function] A quadratic function $f\colon\R^n \to \R$ takes the form: 
+\begin{equation*}
+f(x) = \frac{1}{2}x^T A x - b^T x + c,
+\end{equation*}
+where $A \in \mathbb{S}^n$, $b \in \R^n$ and $c \in \R$.
```
Comment (mrtzh) [inline notes/lecture6.tex]: We haven't use this notation for symmetric matrices, it usually denotes the sphere.

File: notes/lecture6.tex [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+\end{equation*}
+It follows that:
+\begin{align*}
+\left\|x_{t+1} - x^*\right\|_2 &\leq \left(1 - \frac{1}{\kappa}\right)^t \left\|x_1 - x^*\right\|_2 \\
+&\leq \exp \left(-\frac{t}{\kappa}\right) \left\|x_1 - x^*\right\|_2.
+\end{align*}
+\end{proof}
+
+\subsection{Accelerated gradient descent}
+In the previous section, we proved a lower bound on the convergence rate. In this section, we would like to improve on this. Consider whether there was any point where we were careless in the proof above? One obvious candidate is that our choice of step size, $\eta_k = \frac{1}{\beta}$, was chosen rather arbitrarily. In fact, by choosing the sequence $\eta_k$ we can select \textit{any} degree-$t$ polynomial of the form:
```
Comment (mrtzh) [inline notes/lecture6.tex]: it's an upper bound.

File: notes/lecture6.tex [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+Note that:
+% TODO: this is apparently an equality, do we want to strengthen it?
+\begin{equation*}
+\left\|p(A)\right\|_2 \leq \max_{x \in \lambda(A)} \left|p(x)\right|
+\end{equation*}
+% Note this is actually also an equality.
+where $p(A)$ is a matrix polynomial, and $p(t)$ is the corresponding scalar polynomial. In general, we may not know the set of eigenvalues $\lambda(A)$, but we do know that all eigenvalues are in the range $[\alpha, \beta]$. Relaxing the inequality accordingly:
+\begin{equation*}
+\left\|p(A)\right\| \leq \max_{x \in [\alpha, \beta]} \left|p(x)\right|.
+\end{equation*}
```
Comment (mrtzh) [inline notes/lecture6.tex]: Yes, you can make these equalities if you like.

File: notes/lecture6.tex [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+We want a polynomial $p(t)$ that takes on small values in $[\alpha, \beta]$. We impose an additional normalization constraint that $p(0) = 1$ (otherwise we could `cheat' by scaling the polynomial down everywhere on its domain), but allow it to take on arbitrary values elsewhere.
+
+\subsubsection{A naive polynomial solution}
+A naive solution chooses a uniform step size $\eta_t = \frac{2}{\alpha + \beta}$. Note that:
+\[
+\max_{x \in [\alpha, \beta]} \left|1 - \frac{2}{\alpha + \beta} x\right| = \frac{\beta - \alpha}{\alpha + \beta} \leq \frac{\beta}{\alpha} = \kappa,
+\]
+recovering the same convergence rate we proved previously.
+\begin{figure}[h]
+\includegraphics[width=15cm]{figures/lecture6-naive_polynome.pdf}
```
Comment (mrtzh) [inline notes/lecture6.tex]: please specify width as a fraction of `\textwidth`. E.g., `0.9\textwidth` if that's what you want.

File: notes/lecture6.tex (hunk 1 above)
Comment (mrtzh) [reply notes/lecture6.tex]: I believe lecture 1 introduced some notation for this. If not, just say it's symmetric without introducing notation for the set. Thanks.

**Comment by mrtzh** [comment]
Mostly looks great, just a few minor comments before I merge. Thanks.

**Comment by mrtzh** [comment]
Oh, and please add your names to `contributors.tex`.

**Pull request merged**